site/src/components/WorkspacesTable/WorkspacesTableBody.tsx

```diff
@@ -1,98 +1,68 @@
 import Button from "@material-ui/core/Button"
 import Link from "@material-ui/core/Link"
 import { makeStyles } from "@material-ui/core/styles"
-import TableCell from "@material-ui/core/TableCell"
-import TableRow from "@material-ui/core/TableRow"
 import AddOutlined from "@material-ui/icons/AddOutlined"
-import { ChooseOne, Cond } from "components/Conditionals/ChooseOne"
+import { Workspace } from "api/typesGenerated"
+import { TableEmpty } from "components/TableEmpty/TableEmpty"
 import { FC } from "react"
 import { useTranslation } from "react-i18next"
 import { Link as RouterLink } from "react-router-dom"
-import { workspaceFilterQuery } from "../../util/filters"
-import { WorkspaceItemMachineRef } from "../../xServices/workspaces/workspacesXService"
-import { EmptyState } from "../EmptyState/EmptyState"
 import { TableLoader } from "../TableLoader/TableLoader"
 import { WorkspacesRow } from "./WorkspacesRow"
 
 interface TableBodyProps {
-  isLoading?: boolean
-  workspaceRefs?: WorkspaceItemMachineRef[]
-  filter?: string
-  isNonInitialPage: boolean
+  workspaces?: Workspace[]
+  isUsingFilter: boolean
+  onUpdateWorkspace: (workspace: Workspace) => void
 }
 
 export const WorkspacesTableBody: FC<
   React.PropsWithChildren<TableBodyProps>
-> = ({ isLoading, workspaceRefs, filter, isNonInitialPage }) => {
+> = ({ workspaces, isUsingFilter, onUpdateWorkspace }) => {
   const { t } = useTranslation("workspacesPage")
   const styles = useStyles()
 
+  if (!workspaces) {
+    return <TableLoader />
+  }
+
+  if (workspaces.length === 0) {
+    return isUsingFilter ? (
+      <TableEmpty message={t("emptyPageMessage")} />
+    ) : (
```

Comment:
Not distinguishing whether this is the first page makes a search with no results say "No results **on this page**" which I think is a little bit confusing because it implies there are results on other pages but in this case there are actually no results at all (before it would say "No results matched your search").

Reply:
Yes, I saw that in Storybook 😔 I'm going to fix it but I think there is a better way of doing that without using an extra prop. I can be wrong so let's see.